src/xtopcom/xbasic/xhex.h

```diff
@@ -41,6 +41,43 @@ std::string to_hex_prefixed(T const & input) {
     return to_hex(input.begin(), input.end(), "0x");
 }
 
+template <class Iterator>
+std::string to_hex_shrink_0(Iterator begin, Iterator end, std::string const & prefix) {
```

Comment:
we have `std::string to_hex(Iterator begin, Iterator end, std::string const & prefix)` already, so use it directly in `to_hex_shrink_0` and add your shrink leading zero after it.